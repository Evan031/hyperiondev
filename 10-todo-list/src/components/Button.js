import React, {useState} from 'react';
import './Button.css';

function Button({label}) {

    return (
        <div data-testid='button' className='btn btn-primary'>{label}</div>
    );
}

export default Button;
