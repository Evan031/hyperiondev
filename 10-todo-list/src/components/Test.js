import React, {useState} from 'react';
import './Test.css';
import { sqrt, evaluate } from 'mathjs'



function Test() {

    // let foo = '2+3';

    return (
        // <form onSubmit={formSubmitHandler} className="mb-5">
        //     <label htmlFor="name">Calculator:</label>
        //     {/* prettier-ignore */}
        //     <input type="text" className="form-control mb-2" id="calc" name="calc" onChange={InputChangeHandler}  />

        //     {/* prettier-ignore */}
        //     <button type="submit" className="btn btn-primary">Submit</button>
        // </form>
        <p>{evaluate('1500*30%')}</p>
    );
}

export default Test;
