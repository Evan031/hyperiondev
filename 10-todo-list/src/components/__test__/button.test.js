import React from 'react';
import {ReactDOM} from 'react';
import Button from '../Button';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom';
import Test from '../Test';

describe('Button', () => {
    it('renders without crashing', () => {
        const {container} = render(<Button></Button>);
        expect(container).toMatchSnapshot();
    });

    it('renders button correctly', () => {
        const {getByTestId} = render(<Button label="click me please"></Button>);
        expect(getByTestId('button')).toHaveTextContent('click me please');
    });
});

// Test('properly adds two numbers', () => {
//     expect(sum(1, 2)).toBe(3);
// });

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
