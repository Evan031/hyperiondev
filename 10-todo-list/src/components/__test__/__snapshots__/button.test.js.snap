// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders the component 1`] = `
<div>
  <div
    class="btn btn-primary"
    data-testid="button"
  />
</div>
`;

exports[`Button renders without crashing 1`] = `
<div>
  <div
    class="btn btn-primary"
    data-testid="button"
  />
</div>
`;
